adding `WithCustomTag` to a few libs in `contrib/`: `gorm/gorm.v1`, `confluentinc/confluent-kafka-go/kafka`, `database/sql`, `google.golang.org/grpc`

duxing opened the pull request:
support custom tags for a few libs in `contrib/`: 
 - `gorm.io/gorm.v1`
 - `confluentinc/confluent-kafka-go/kafka`
 - `database/sql`
 - `google.golang.org/grpc`

Fixes #1308

ajgajg1134 commented on contrib/gorm.io/gorm.v1/option.go:
```diff
@@ -71,3 +74,19 @@ func WithErrorCheck(fn func(err error) bool) Option {
+		if tagFn != nil {
+			cfg.tagFns[tag] = tagFn
+		} else {
+			delete(cfg.tagFns, tag)
```
I'm curious why you went with a slightly different approach here compared to the other `WithCustomTag` function? Do you foresee needing to be able to remove/delete tagFns after they've already been added?

duxing replied:
tbh I don't think it's practically useful. added this implementation to behave in a way that callers would assume. Not a must-have and I can remove it if you prefer.

ajgajg1134 reviewed (changes requested):
Hi duxing! Thanks for this contribution, it looks like a good idea to bring both these versions to feature parity. Just one small comment I've attached, and also can you also add a unit test for this change? You may be able to borrow nearly all the test from the previous version which has a test `TestCustomTags` that covers this.

Thanks again!

duxing commented:
hi @ajgajg1134  sorry about the delay. I was able to get to this last night and added test coverage for everything I introduced. can you take another look when you have a chance?

I noticed the CI fails complaining about `DD_API_KEY` missing in the env (and another `bitnami/kafka` related issue). I'm not sure how to address these ones, it'll be helpful if you can share some insights with me

ajgajg1134 commented:
> hi @ajgajg1134 can you help me understand why my PR fails due to:
> 
> > Neither DATADOG_API_KEY nor DD_API_KEY is in your environment.
> > Internal Error: API key is missing
> > when other PRs (eg. [this](https://github.com/DataDog/dd-trace-go/pull/1348) and [this](https://github.com/DataDog/dd-trace-go/pull/1267) ) don't have the problem? it's probably something trivial and I'd like to figure out what I missed.
> 
> thanks in advance!

Hello @duxing! Sorry, the system-test failures you're seeing are due to this PR running a branch from an external fork of the repository. Unfortunately it means these tests can't be run unless the branch is coming from within the repository, although we can run the other tests still and should be able to merge after review is complete. (This is a limitation of github actions that hopefully is resolved at some point)

duxing commented:
@ajgajg1134 thx for explaining, i figured it has something to do with the github action settings. I've disabled my own fork since it's complaining about the same thing.

is it possible to cherry-pick my change to a branch for this repo and proceed?

ajgajg1134 commented:
Closing this PR since the work was merged with #1359
